### PR TITLE
Breaking object sources

### PR DESCRIPTION
**Pull request by tiennou**
This is the breaking part of #2255. Settings view will go missing because the 1st commit, and plugins _will_ fail to load because of 0e22a18 (at least not the core ones. I think).

### CODE REVIEW

File: Quicksilver/Code-QuickStepCore/QSObjectSource.h [hunk 1]
```diff
@@ -112,7 +107,5 @@
 // Please use -selectedEntry instead of those
 // The rational being that between -currentEntry, -selection and direct Ivar
 // everything can go wrong.
```
Comment (pjrobertson) [inline Quicksilver/Code-QuickStepCore/QSObjectSource.h]: Wait, I mentioned these in your last PR I just looked at, saying "why not just remove them?". The answer seems to be dyld right? But here you've decided to remove them?!

Perhaps the comment needs removing as well?

**Comment by pjrobertson** [comment]
OK I see what you're doing here, splitting up the breaking stuff and non-breaking stuff. Just to be sure: this PR would need to be merged *before* #2255 is merged right?

**Comment by tiennou** [comment]
The reverse actually. This one's on top of #2255, after we went through object sources.

File: Quicksilver/Code-QuickStepCore/QSObjectSource.h (hunk 1 above)
Comment (tiennou) [inline Quicksilver/Code-QuickStepCore/QSObjectSource.h]: Oops ! dyld is effectively quite good at catching disappearing ivars...

**Comment by skurfer** [comment]
So shouldn't the NIBs be updated too? For example, the `QSFileSystemObjectSource.xib` is still using `selection.info.settings`. That should change to `selectedEntry.sourceSettings`, right?

**Comment by josephholsten** [comment]
with #2255 merged, does this need to exist any longer?

**Comment by skurfer** [comment]
I believe that was part 1, the backward-compatible part. This is part 2, which could break stuff.

**Comment by n8henrie** [comment]
It's been several years, I'm going to close this as stale, happy to reopen if work on this resumes! Thank you @tiennou for your efforts here!

**Pull request closed without merging**